Makefile.osx

```diff
@@ -1,12 +1,18 @@
 include Makefile.srcs
 
-CC = gcc
-CXX = g++
+CC = clang
+CXX = clang
 AR = ar
 
-CXXFLAGS = -O2 -arch x86_64 -mmacosx-version-min=10.11
-LDFLAGS = -arch x86_64 -lstdc++ -mmacosx-version-min=10.11
-PLATFORM = OSXx64
+ifeq ($(BUILD_TARGET),P4PluginX64)
+	PLATFORM = OSXx64
+	CXXFLAGS = -O2 -arch x86_64 -mmacosx-version-min=10.11
+	LDFLAGS = -L./P4Plugin/Source/p4api/lib/osx64 -L./P4Plugin/Source/openssl/lib/osx64 -arch x86_64 -lstdc++ -mmacosx-version-min=10.11
+else
+	PLATFORM = OSXarm64
+	CXXFLAGS = -O2 -arch arm64 -mmacosx-version-min=12.0 
```

Comment:
Just want to check on one thing, we may have raised the min spec for the editor in 6.3+ to macOS 13.0. If that is true then we should change the arg to all the `-mmacosx-version-min` flags to 13.0. I'm checking and I'll get back to you here

Reply:
Oh sorry, just saw this comment now. In that case keep the min spec as is, except for the ones I changed to 12.0, I added suggestions to put that back to 11.0.